### PR TITLE
Add the limits library for rate limiting implementation

### LINKED ISSUE #353: [FEATURE] Rate limiting algorithms
**Is your feature request related to a problem? Please describe.**
I've recently added rate limit support to Astra upserts, which uses `ratelimit` package, but that one looks a bit abandoned. It hasn't been updated for 5 years and it is unclear how it actually works.

**Describe the solution you'd like**
I'd like to have our own `@ratelimit` decorator which implements various rate limiting algos. Start with ones that do not require external storage.

### PR DESCRIPTION
**Pull request by s-vitaliy**
Resolves #353

## Scope

Implemented:
 - Adopted the `limits` library for rate limiting
 - Added the `rate_limit` decorator.

Additional changes:
- The `rate_limit_calls` and `rate_limit_period_seconds` parameters in the AstraClient replaced with new human-readable parameter `client_rate_limit` (e. g. `client_rate_limit= "1000 per second"`).
- Default rate limiting algorithm changed from fixed window to moving window.

## Checklist

- [x] GitHub issue exists for this change.
- [x] Unit tests added and they pass.
- [x] Pylint 10.0/10.0 without bloating `.pylintrc` with exceptions.
- [x] Review requested on `latest` commit.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -17,6 +17,7 @@ cryptography = "~36.0"
 pandas = { version = ">=2.0.0,<3.0", extras = ["performance"] }
 pyarrow = ">=7.0"
 dataclasses-json = "~0.6"
+limits = "~3.7"
```
Comment (george-zubrienko) [inline pyproject.toml:20]: I'd suggest we replace the one we have right away, wdyt? And bump minor

File: adapta/utils/decorators/_rate_limit.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+def rate_limit(
+    _func: Callable = None,
+    *,
+    limit: str,
+    strategy: str = "moving-window",
+    delay_func: Callable[[], int] = _default_delay_func
+) -> Callable:
+    """
+    Rate limit decorator.
+    :param limit: the limit string to parse
```
Comment (george-zubrienko) [inline adapta/utils/decorators/_rate_limit.py:27]: Examples please :)

File: adapta/utils/decorators/_rate_limit.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    :param strategy: the strategy to use
```
Comment (george-zubrienko) [inline adapta/utils/decorators/_rate_limit.py:28]: Can we have an enum here?

**Review by george-zubrienko — APPROVED** [review]
Awesome stuff!

File: adapta/utils/decorators/_rate_limit.py (hunk 3 above)
Comment (s-vitaliy) [reply adapta/utils/decorators/_rate_limit.py:28]: No, the caller code should use the same names as the `limits` library uses. If we define our own enum, we may have a compatibility issues.

Comment (s-vitaliy) [reply adapta/utils/decorators/_rate_limit.py:28]: Discussed by voice: will do an enum.

**Pull request merged**